File: ProjectWe.Models/Project.cs
using ProjectWe.Models.Interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProjectWe.Models
{
    public class Project: IAuditableEntity
    {
        public int ProjectId { get; set; }
        public string Name { get; set; } = null!;
        public string Description { get; set; } = null!;
        public DateTime? CreatedAt { get; set; }
        public DateTime? LastModified { get; set; }
        public int UserId { get; set; }
        public int StatusId { get; set; }
        public string Status { get; set; } = null!;
        public int CityId { get; set; }
        public string City { get; set; } = null!;
        public int CategoryId { get; set; }
        public string Category { get; set; } = null!;
    }
}

File: projectwe_mobile/lib/src/features/projects/models/project.dart
import 'package:json_annotation/json_annotation.dart';

part 'project.g.dart';

@JsonSerializable()
class Project {
  int? projectId;
  String name;
  String description;
  String city;
  int cityId;
  String category;
  int categoryId;
  String status;
  int statusId;

  Project({
    this.projectId,
    this.name = "",
    this.description = "",
    this.city = "",
    this.cityId = 0,
    this.category = "",
    this.categoryId = 0,
    this.status = "",
    this.statusId = 0,
  });

  factory Project.fromJson(Map<String, dynamic> json) => _$ProjectFromJson(json);

  /// Connect the generated [_$ProjectToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$ProjectToJson(this);
}